hwk-gen.py

```diff
@@ -0,0 +1,168 @@
+            sanitized_dates.append(day)
+
+    return sanitized_dates
+
+def day_to_num(day):
+    if day == "M":
+        return 0
+    elif day == "T":
+        return 1
+    elif day == "W":
+        return 2
+    elif day == "R":
+        return 3
+    elif day == "F":
+        return 4
+    else:
+        raise ValueError("day_to_num should contain only chars from" +
+                         legal_days_of_week)
+
+
+
+
+def make_date_list(start_date, weeks, days_of_week, holiday_list):
+    """
+    return list of date strings in yyyy-mm-dd format
+
+    Example:
+
+    TODO shouldn't the 3 below be 2? As this is for 2 weeks?
+     make_date_list("2019-03-31",3,"MW",["2019-04-08","2019-04-09"]) =>
+       ["2019-04-01","2019-04-03","2019-04-10"]
+    """
+
+    validate_days_of_week(days_of_week)
+    start_datetime = parse(start_date)
+
+    final_datetimes = []
+    for i in range(weeks):
+        start_of_week_datetime = add_weeks(start_datetime,i)
+        days_this_week_unsanitized = days_for_this_week(start_of_week_datetime, days_of_week)
+        days_this_week_sanitized = days_without_holidays(days_this_week_unsanitized, holiday_list)
+        final_datetimes.append(days_this_week_sanitized)
+
+    return final_datetimes
+
+
+def hwk_gen(path, start_date, weeks, assign_day_of_week, days_till_due, holiday_list):
+    """
+    Creates a _hwk directory in the given path with premade hwk stubs
+    according to the other fields:
+    start date: Must be a Sunday, day that the quarter or semester starts
+    weeks: number of weeks that the class goes for (ex 10 for quarter)
+    assign day of week: day of week that homework is assigned on
+    days till due: number of days ahead that each hw is due (number of days students get to work on it)
+    list of holidays: days that homeworks can't be assigned on
+    """
+    #Create path:
+    directory_path = os.path.join(path, "_hwk")
+    try:
+        os.makedirs(directory_path)
+    except FileExistsError:
+        print ("directory already exists error: Creation of the directory %s failed" % directory_path)
+        raise
+    except OSError:
+        print ("OS error: Creation of the directory %s failed" % directory_path)
+        raise
+    else:
+        print ("Successfully created the directory %s" % directory_path)
+
+    #create valid dates listing:
+    valid_dates = make_date_list(start_date, weeks, days_of_week, holiday_list)
+    lecture_num = 0
+    for dates_by_week in valid_dates:
+        for date in dates_by_week:
+            lecture_num += 1 #first lecture num will be 1
+            filename = "lecture" + str(lecture_num)
+            f = open(os.path.join(directory_path, filename), "w+")
+            f.write("---\n")
+            f.write("num: " + '"lect' + str(lecture_num) + '"\n')
+            f.write("lecture_date: " + str(date.date()) + "\n")
+            f.write("desc: " + '\n')
+            f.write("ready: " + "false\n")
+            f.write("pdfurl: " + "\n")
```

Comment:
`pdfurl` should perhaps be an empty string?
would be good to add a bool flag to the config for whether to include the pdf, in case people don't want/use slides (i.e., if flag is false, then the variable is not included in the front matter).

Reply:
Again, what do you mean by an empty string?
Adding the bool flag now!

Reply:
Changed pdfurl to empty string